zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/schema/Schema.scala

```diff
@@ -184,19 +184,31 @@ trait Measurement {
 
 //TODO when UI get the schema, it should know which is dimension/measure, functions that can apply onto it, etc.
 // so that it won't ask for a inapplicable function such that get the max to a string field
-case class Schema(typeName: String,
-                  dimension: Seq[Field],
-                  measurement: Seq[Field],
-                  primaryKey: Seq[Field],
-                  timeField: TimeField
-                 ) {
+abstract class Schema(typeName: String,
+                      dimension: Seq[Field],
+                      measurement: Seq[Field],
+                      primaryKey: Seq[Field]
+                     ) {
```

Comment:
Yes. The thing is that, if to make a flawless design after introducing static schema, we need to change the `DataSetInfo` and then make previous function calls only available for temporal schema or `DatasetInfo` who contains a temporal dataset. Need to consider carefully.